backends/ebpf/psa/externs/ebpfPsaRandom.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2022-present Orange
+Copyright 2022-present Open Networking Foundation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "ebpfPsaRandom.h"
+
+namespace EBPF {
+
+EBPFRandomPSA::EBPFRandomPSA(const IR::Declaration_Instance* di) :
+        minValue(0), maxValue(0), range(0) {
+    CHECK_NULL(di);
+
+    // verify type
+    if (!di->type->is<IR::Type_Specialized>()) {
```

Comment:
but this whole sequence could be a utility function; you do this a lot in this backend.